### PR TITLE
[Core][Structural] Make RVE 2D compatible

### PR DESCRIPTION
**Pull request by rubenzorrilla**
**📝 Description**
This amends the `RVEPeriodicityUtility` to be 2D compatible. A 2D test similar to the 3D original one in the `StructuralMechanicsApplication` has been added as well.

@bcsnvs @ryzhakov82 I think we can work with this branch while this is merged.

### CODE REVIEW

**Comment by rubenzorrilla** [comment]
Just mentioning that the 2D test has been created by @bcsnvs (merits to him).

File: applications/StructuralMechanicsApplication/python_scripts/rve_analysis.py [hunk 1]
```diff
@@ -15,11 +15,14 @@ def __init__(self, model, project_parameters):
         self.perturbation = project_parameters["rve_settings"]["perturbation"].GetDouble()
 
         self.averaging_volume = -1.0  # it will be computed in initialize
-        domain_size = project_parameters["solver_settings"]["domain_size"].GetInt()
-        if(domain_size == 2):
+        self.domain_size = project_parameters["solver_settings"]["domain_size"].GetInt()
+        if(self.domain_size == 2):
```
Comment (loumalouomega) [inline applications/StructuralMechanicsApplication/python_scripts/rve_analysis.py]:
```suggestion
        if self.domain_size == 2:
```
More pythonic

File: applications/StructuralMechanicsApplication/python_scripts/rve_analysis.py [hunk 2]
```diff
@@ -254,17 +265,17 @@ def _ComputeEquivalentStress(self, i, j, perturbation, boundary_mp, averaging_mp
...
-        strain = KratosMultiphysics.Matrix(3, 3)
+        strain = KratosMultiphysics.Matrix(self.domain_size, self.domain_size)
         strain.fill(0.0)
 
         strain[i, j] = perturbation
         strain[j, i] = perturbation
 
         strain_vector = KratosMultiphysics.Vector(self.strain_size)
-        if(self.strain_size == 2):
+        if(self.strain_size == 3):
```
Comment (loumalouomega) [inline applications/StructuralMechanicsApplication/python_scripts/rve_analysis.py]:
```suggestion
        if self.strain_size == 3:
```

File: applications/StructuralMechanicsApplication/python_scripts/rve_analysis.py [hunk 3]
```diff
@@ -254,17 +265,17 @@ def _ComputeEquivalentStress(self, i, j, perturbation, boundary_mp, averaging_mp
...
             strain_vector[0] = strain[0, 0]
             strain_vector[1] = strain[1, 1]
-            strain_vector[2] = 2.0*strain[1, 2]
+            strain_vector[2] = 2.0*strain[0, 1]
         elif(self.strain_size == 6):
```
Comment (loumalouomega) [inline applications/StructuralMechanicsApplication/python_scripts/rve_analysis.py]:
```suggestion
        elif self.strain_size == 6:
```

File: kratos/utilities/rve_periodicity_utility.h [hunk 4]
```diff
@@ -193,6 +194,14 @@ class KRATOS_API(KRATOS_CORE) RVEPeriodicityUtility
         const Vector& rTranslationVector
         );
 
+    template<std::size_t TDim>
```
Comment (loumalouomega) [inline kratos/utilities/rve_periodicity_utility.h:197]: Doxygen?

**Review by loumalouomega — CHANGES_REQUESTED** [review]
Minor comments

File: kratos/utilities/rve_periodicity_utility.h (hunk 4 above)
Comment (rubenzorrilla) [reply kratos/utilities/rve_periodicity_utility.h:197]: I don't think it is needed in this case as it is a `private` function to be specialized. Doxygen is already present in the `AssignPeriodicity`.

**Review by loumalouomega — APPROVED** [review]

**Pull request merged**